x/twap/logic.go

```diff
@@ -47,15 +48,21 @@ func getSpotPrices(
 	sp0, err0 := k.CalculateSpotPrice(ctx, poolId, denom0, denom1)
 	sp1, err1 := k.CalculateSpotPrice(ctx, poolId, denom1, denom0)
 	if err0 != nil || err1 != nil {
+		previousRecord, err := keeper.getRecordAtOrBeforeTime(ctx, poolId, previousErrorTime, denom0, denom1)
 		latestErrTime = ctx.BlockTime()
-		// In the event of an error, we just sanity replace empty values with zero values
-		// so that the numbers can be still be calculated within TWAPs over error values
-		// TODO: Should we be using the last spot price?
 		if (sp0 == sdk.Dec{}) {
```

Comment:
IMO, with so much if and else cases here, I think this reduces code readability. What do you think of separating out the cases for err0 != nil and err1!= nil?